QWeb/internal/element.py

```diff
@@ -360,7 +367,7 @@ def _calculate_closest_distance(element1: WebElement, element2: WebElement) -> f
     search_direction = CONFIG["SearchDirection"]
```

Comment:
i think it would be better to strip the possible exclamation mark and leave the if statements as they were.
In order to be sure what the search_directions can actually be one will need to find the validation method in search_strategy.py

```
search_direction = CONFIG["SearchDirection"]
# check and strip enforce char
if search_direction.endswith("!")
    search_direction = search_direction[:-1]
```

Reply:
Not sure I get what you mean. the idea here is to actually fail if text is found but not from the direction the anchor (in enforced mode). Previously without enforced mode it did not fail, but selected the "closest". 

Thus we need different handling for enforced and normal mode. Also I want it to be visible in any logging that enforced mode was used -> not wise to strip the "!"

Reply:
The thing I was wondering when looking at the changes was "can values like 'something left and else' get passed in here" as they would pass the checks `if 'left' in search_direction:`

I thought about the stripping after the review and I'd actually prefer making the checks more explicit (instead of stripping) in this fashion:
`if search_direction in ('left', 'left!'):`

I'd also like to see a raising `else:` block with comment that the execution should never get there just to be sure and to remove doubts of the behavior in case if the code reader does not happen to look at the config input validation function.

Reply:
ok, that makes sense.

Reply:
the else branch is not really needed because value is checked when set and it can't be any other than one of those allowed values.

```
SetConfig                   SearchDirection     something left
! keyword: SetConfig                   SearchDirection     something left
! handler execution failed: ValueError: Wrong search direction
```